### PR TITLE
Add merge request support for GitLab repositories

### PR DESCRIPTION
**Pull request by ZuGao**
Resolves FILO-338

Added support for Merge Requests. Unfortunately no unit tests in this MR as there wasn't enough time to implement mocks even though I went ahead and did some development. I will create another commit for the unit tests

### CODE REVIEW

File: cmd/tune.go [hunk 1]
```diff
@@ -25,6 +25,9 @@ func init() {
 	tuneCmd.Flags().BoolP("es-insecure-skip-tls-verify", "k", config.ElasticSearch.InsecureSkipVerify, "If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure")
 	tuneCmd.Flags().String("es-custom-ca", config.ElasticSearch.CustomCA, "Custom CA certificate to trust (in PEM format)")
 	tuneCmd.Flags().String("es-custom-ca-file", config.ElasticSearch.CustomCAFile, "Path to custom CA certificate to trust (in PEM format)")
+	tuneCmd.Flags().StringP("repository", "r", config.GitLab.Repository, "Repository Local file system path or a URL")
+	tuneCmd.Flags().StringP("gitlab-token", "t", config.GitLab.Token, "GitLab access token")
+	tuneCmd.Flags().BoolP("create-merge-request", "m", config.GitLab.MergeRequest, "Flag to create or not a merge request")
```
Comment (bittner) [inline cmd/tune.go:30]: I'd rather use `-c` here, for "create", but that's certainly personal taste. The help text is a bit confusing.

```suggestion
	tuneCmd.Flags().BoolP("create-merge-request", "c", config.GitLab.MergeRequest, "Create a merge request on GitLab")
```

File: pkg/file/file.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	_, err := f.File.Write([]byte(content))
+	if err != nil {
+		return fmt.Errorf("cannot write to file %s: %w", f.File.Name(), err)
+	}
+	return nil
+}
+
+func (f *RuleFile) Close() error {
+	err := f.File.Close()
+	if err != nil {
```
Comment (bittner) [inline pkg/file/file.go:49]: Personally, I find the "all in one line" style appealing (applies to all `if`s in this PR). I think it reduces visual clutter.

Can you combine the two lines above into a single one? e.g.

```suggestion
	if err := f.File.Close(); err != nil {
```

Maybe there is a reason why you would not want to do that. Let me know.

File: pkg/file/file_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	assert.Truef(t, file!=nil, "there should be a file opened")
+}
+
+func TestGet(t *testing.T) {
+
+
+
+
+	//mockFile.EXPECT().Read(gomock.Any()).Return(0, errors.New("EOF"))
+
```
Comment (bittner) [inline pkg/file/file_test.go:40]: I would remove this function if it doesn't do anything yet. Or at least clean it up.

Comment (ZuGao) [reply pkg/file/file_test.go:40]: yeah that was a leftover I will update it

File: cmd/tune.go (hunk 1 above)
Comment (ZuGao) [reply cmd/tune.go:30]: I left `-m` as written in the specifications

File: pkg/file/file_test.go (hunk 3 above)
Comment (ZuGao) [reply pkg/file/file_test.go:40]: I temporarily deleted this file

**Comment by bittner** [comment]
```console
$ go fmt ./...
cfg/types.go
pkg/file/file.go
pkg/git/git.go
pkg/tuner/tuner.go
```
```console
$ git status
On branch feature/merge-request
Your branch is up to date with 'origin/feature/merge-request'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   cfg/types.go
        modified:   pkg/file/file.go
        modified:   pkg/git/git.go
        modified:   pkg/tuner/tuner.go
```
There is some minor formatting that you should correct.

I would suggest you copy the Seiso project's [Makefile](https://github.com/appuio/seiso/blob/master/Makefile) over here and remove the bits unneeded in this project. That's a quick thing to do.  After that it should be natural to run `make` or `make fmt` before pushing to a PR.

**Comment by ZuGao** [comment]
Introduced Makefile, formatted the code

**Pull request merged**